AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -187,91 +192,186 @@ + (void)deleteDocumentWithPartition:(NSString *)partition
 }
 
 #pragma mark - MSDataStore Implementation
+
+- (void)replaceWithPartition:(NSString *)partition
+                  documentId:(NSString *)documentId
+                    document:(id<MSSerializableDocument>)document
+                writeOptions:(MSWriteOptions *)writeOptions
+           completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler {
+  
+  // In the current version we do not support E-tag optimistic concurrency logic and `replace` will call Create (POST) operation instead of Replace (PUT).
+  [self createWithPartition:partition
+                 documentId:documentId
+                   document:document
+               writeOptions:writeOptions
+          completionHandler:completionHandler];
+}
+
+- (void)readWithPartition:(NSString *)partition
+               documentId:(NSString *)documentId
+             documentType:(Class)documentType
+              readOptions:(MSReadOptions *)__unused readOptions
+        completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler {
+  
+  // TODO consume writeOptions.
+  [MSTokenExchange
+   performDbTokenAsyncOperationWithHttpClient:(MSStorageIngestion *)self.ingestion
+   partition:partition
+   completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull error) {
+     // If error getting token.
+     if (error || !tokenResponses) {
+       MSLogError([MSDataStore logTag], @"Can't get CosmosDb token:%@", [error description]);
+       completionHandler([[MSDocumentWrapper alloc] initWithError:error documentId:documentId]);
+       return;
+     }
+     
+     // Create http client.
+     MSCosmosDbIngestion *cosmosDbIngestion = [[MSCosmosDbIngestion alloc] init];
+     
+     // Call CosmosDb.
+     [MSCosmosDb
+      performCosmosDbAsyncOperationWithHttpClient:cosmosDbIngestion
+      tokenResult:tokenResponses.tokens[0]
+      documentId:documentId
+      httpMethod:@"GET"
```

Comment:
Can you use a constant for GET?

Reply:
@yochaude and I have added one in our PR, it's probably OK for now to avoid conflicts